src/controllers/AuthLocalController.js

```diff
@@ -104,26 +173,122 @@ export default class AuthLocalController {
-      ? res.status(status.OK).send({ message: 'Success! your password has been changed.' })
-      : res.status(status.NOT_MODIFIED).send({ message: 'Password not updated' });
+    const isUpdated = await User.update({ password: helper.password.hash(passwordOne) }, { email });
+    return isUpdated
+      ? res
+        .status(status.OK)
+        .json({ isUpdated, message: 'Success! your password has been changed.' })
+      : res.status(status.NOT_MODIFIED).json({ errors: 'Password not updated' });
+  }
+
+  // follow user
+  /**
+   * @description function to create user follows
+   * @param {object} req request from user
+   * @param {object} res server response
+   * @returns {object} true
+   */
+  static async follow(req, res) {
+    const { username } = req.params;
+    const checkUser = await User.findOne({ username });
+    if (checkUser.id === req.user.id) {
+      return res
+        .status(status.BAD_REQUEST)
+        .json({ errors: { follow: 'You can not follow your self ' } });
+    }
+    const follow = await User.follow.add({
+      followed: checkUser.id,
+      userId: req.user.id
+    });
+    if (follow.errors) {
+      return follow.errors.name === 'SequelizeUniqueConstraintError'
+        ? res
+          .status(status.EXIST)
+          .send({ errors: { follow: `You are already following "${username}"` } })
+        : res.status(status.SERVER_ERROR).json({ errors: 'oops, something went wrong' });
+    }
+    return res.status(status.CREATED).json({
+      message: `now you are following ${checkUser.username}`
+    });
+  }
+
+  // unFollow user
+  /**
+   * @description function to allow user to unfollow users
+   * @param {object} req user request
+   * @param {object} res response from server
+   * @returns {object} true
+   */
+  static async unfollow(req, res) {
+    const [username, user] = [req.params.username, req.user];
+    const checkUser = await User.findOne({ username });
+
+    const hasUnfollowed = Object.keys(checkUser).length
+      ? await User.follow.remove({ userId: user.id, followed: checkUser.id })
+      : null;
+
+    if (hasUnfollowed && hasUnfollowed.errors) {
+      return res.status(status.SERVER_ERROR).json({ errors: 'oops, something went wrong!!' });
+    }
+    return hasUnfollowed
+      ? res.status(status.OK).json({
+        message: `you unfollowed ${username}`
+      })
+      : res
+        .status(status.BAD_REQUEST)
+        .json({ errors: { follow: `you are not following "${username}"` } });
+  }
+
+  /**
+   * @description function to fetch users'followers
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} followers
+   */
+  static async followers(req, res) {
+    const { id } = req.user;
+    const followers = await User.follow.getAll({ followed: id });
+    return followers.length
+      ? res.status(status.OK).json({
+        message: 'Followers',
+        followers: followers.map(follower => delete follower.get().followedUser && follower)
```

Comment:
why are we deleting?

Reply:
I remove the followed user here because I  wont to display only the user that follows me in this controller and in the model I get both followers and following